c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -790,6 +791,13 @@ cudaError_t cudaMallocMaybeCapturing(void** p, size_t size) {
 
 static std::string reportProcessMemoryInfo(int device) {
 #ifdef PYTORCH_C10_DRIVER_API_SUPPORTED
+  static c10::once_flag once;
+  static void* handle_1;
+  c10::call_once(
+      once, [] {handle_1 = dlopen("libnvidia-ml.so.1", RTLD_LAZY);});
```

Comment:
nit: shouldn't this be factored into a `init_driver_api()` function in the driver API file that internally will handle the call_once and can be called from any place where it might need to get loaded. That will avoid potential duplicate dlopen as well as scattered dlopen code.

Reply:
Good idea, fixed by moving dlopen into DriverAPI::get_nvml_handle() in newest refactor